router/batchrouter/asyncdestinationmanager/common/common.go

```diff
@@ -0,0 +1,163 @@
+package common
+
+import (
+	stdjson "encoding/json"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/tidwall/gjson"
+
+	"github.com/rudderlabs/rudder-go-kit/config"
+	backendconfig "github.com/rudderlabs/rudder-server/backend-config"
+	"github.com/rudderlabs/rudder-server/jobsdb"
+	"github.com/rudderlabs/rudder-server/services/rsources"
+)
+
+type AsyncDestinationManager interface {
+	Upload(asyncDestStruct *AsyncDestinationStruct) AsyncUploadOutput
+	Poll(pollInput AsyncPoll) PollStatusResponse
+	GetUploadStats(UploadStatsInput GetUploadStatsInput) GetUploadStatsResponse
+}
+
+var AsyncDestinations = []string{"MARKETO_BULK_UPLOAD", "BING_ADS"}
+
+type PollStatusResponse struct {
+	Complete       bool
+	InProgress     bool
```

Comment:
Any reason why we would need both Complete and InProgress?

Reply:
yes, we will need this. BingAds sends back status for inprogress processes. We are handling that here